src/kikuchipy/indexing/_refinement/_objective_functions.py

```diff
@@ -164,17 +166,19 @@ def _refine_orientation_pc_objective_function(x: np.ndarray, *args) -> float:
     -------
         normalized cross-correlation score.
     """
-    dc = _get_direction_cosines_for_fixed_pc(
-        pcx=x[3],
-        pcy=x[4],
-        pcz=x[5],
-        nrows=args[7],
-        ncols=args[8],
+    det = EBSDDetector(
```

Comment:
Should I have a look at speeding up this computation?

Reply:
Thanks, but I think I've already fixed it. I found a way to pass the information through without instantiating the EBSDDetector. Updates coming today or tomorrow hopefully.

Reply:
I managed to optimise my branch so that it is now the same speed as the develop branch:

```python
>>> import kikuchipy as kp
>>> s = kp.data.nickel_ebsd_small()
>>> mp = kp.data.nickel_ebsd_master_pattern_small(projection="lambert")
>>> _ = s.refine_orientation_projection_center(s.xmap, s.detector, mp, 20)

# This PR (before opt.)
>>> %timeit s.refine_orientation_projection_center(s.xmap, s.detector, mp, 20)
1.57 s ± 13.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

# This PR OPTIMISED
>>> %timeit s.refine_orientation_projection_center(s.xmap, s.detector, mp, 20)
534 ms ± 9.98 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

# develop branch
>>> %timeit s.refine_orientation_projection_center(s.xmap, s.detector, mp, 20)
542 ms ± 10.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```